pallets/delegation/src/lib.rs

```diff
@@ -389,12 +397,80 @@ impl<T: Trait> Module<T> {
 	}
 }
 
+#[derive(Encode, Decode)]
+pub struct Delegation<T: Trait> {
+	pub root_id: T::DelegationNodeId,
+	pub parent: Option<T::DelegationNodeId>,
+	pub owner: T::AccountId,
+	pub permissions: Permissions,
+	pub revoked: bool,
+}
+
+impl<T: Trait> Delegation<T> {
+	pub fn new_root(
+		root_id: T::DelegationNodeId,
+		owner: T::AccountId,
+		permissions: Permissions,
+	) -> Self {
+		Delegation {
+			root_id,
+			owner,
+			permissions,
+			revoked: false,
+			parent: None,
+		}
+	}
+
+	/// new_child creates a new child node for the delegation tree.
+	///
+	/// root_id - the root of the delegation tree
+	/// parent - the parent in the tree
+	/// owner - the owner of the new child root. He will receive the delegated permissions
+	/// permissions - the permissions that are delegated
+	pub fn new_child(
+		root_id: T::DelegationNodeId,
+		parent: T::DelegationNodeId,
+		owner: T::AccountId,
+		permissions: Permissions,
+	) -> Self {
+		Delegation {
+			root_id,
+			parent: Some(parent),
+			owner,
+			permissions,
+			revoked: false,
+		}
+	}
+}
+
+#[derive(Encode, Decode)]
+pub struct DelegationProperties<T: Trait> {
+	pub ctype_hash: T::Hash,
+	pub account: T::AccountId,
```

Comment:
Maybe we should add a comment which displays what this field stands for? `account` is very undescriptive.

Reply:
yes. It's the owner right? Wasn't sure because the delegation root also has an owner.

Reply:
I am not a 100% sure either, but that's what I assumed.

Reply:
Checked the whitepaper to be sure: It is indeed the owner.

Reply:
fixed in 951c5da